rdflib/store.py

```diff
@@ -226,14 +226,18 @@ def addN(self, quads):
         statement is quoted/hypothetical. Note that the default implementation
         is a redirect to add
         """
+        count = 0
         for s, p, o, c in quads:
-            assert c is not None, "Context associated with %s %s %s is None!" % (
-                s,
-                p,
-                o,
-            )
-            self.add((s, p, o), c)
-
+            assert c is not None, \
+                "Context associated with %s %s %s is None!" % (s, p, o)
+            added = self.add((s, p, o), c)
+            if (isinstance(added,int)):
```

Comment:
Stores that do not support count triples should not be forced to deal with this branch. Should probably dispatch on whether or not the store supports returning counts.

Reply:
Stores that do not support count triples would not return anything from their `add `method, in which case `None` will be returned from this (i.e. `addN`) method. I could not understand how this would be an issue?

Reply:
Since there hasn't been any movement on this: I was advocating for an explicit tag on stores to know if a count is returned, but at this point, we weren't using the return value of add/addN for anything, so until another feature comes along, this should be fine.

As far as "dealing with the branch", I assume you're thinking of performance based on your general comment, but this isn't a tight loop since we're making a function call, so there's barely in additional cost here.